### PR TITLE
filters/auth: add -oauth2-token-cookie-remove-subdomains flag

### PR DESCRIPTION
**Pull request by AlexanderYastrebov**
See related https://github.com/zalando/skipper/issues/1909

Signed-off-by: Alexander Yastrebov <alexander.yastrebov@zalando.de>

### CODE REVIEW

File: filters/auth/grantconfig.go [hunk 1]
```diff
@@ -91,6 +92,11 @@ type OAuthConfig struct {
 	// encrypted access token after a successful token exchange.
 	TokenCookieName string
 
+	// TokenCookieRemoveSubdomains sets the number of subdomains to remove from
+	// the callback request hostname to obtain token cookie domain.
+	// Init converts default nil to 1.
+	TokenCookieRemoveSubdomains *int
```
Comment (AlexanderYastrebov) [inline filters/auth/grantconfig.go:98]: To stay backwards-compatible I had to use pointer here to tell zero and default value (one) apart.
The proper config initialization requires `Init()` call.

If we can sacrifice a bit of compatibility for library users then we could use just `int` and rely on the default skipper flag value of 1.

Comment (AlexanderYastrebov) [reply filters/auth/grantconfig.go:98]: There are not many options here https://stackoverflow.com/questions/68800319/how-to-differentiate-int-null-and-defaulted-to-zero-from-int-actually-equal-to-z

We could also create a wrapper type but that feels too much.

Comment (szuecs) [reply filters/auth/grantconfig.go:98]: If you need this 3 state: 0 value, 0, N, then ptr is the way to go

File: filters/auth/grant_test.go [hunk 2]
```diff
@@ -214,14 +217,22 @@ func checkStatus(t *testing.T, rsp *http.Response, expectedStatus int) {
 	}
 }
 
-func checkRedirect(t *testing.T, rsp *http.Response, expectedURL string) {
+func checkRedirect(t *testing.T, rsp *http.Response, expectedLocationWithoutQuery string) {
```
Comment (szuecs) [inline filters/auth/grant_test.go:220]: expectedURI maybe?

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**